src/block/put.js

```diff
@@ -3,35 +3,72 @@
 const promisify = require('promisify-es6')
 const Block = require('ipfs-block')
 const CID = require('cids')
-const once = require('once')
+const multihash = require('multihashes')
 const SendOneFile = require('../utils/send-one-file')
 
 module.exports = (send) => {
   const sendOneFile = SendOneFile(send, 'block/put')
 
-  return promisify((block, cid, _callback) => {
-    // TODO this needs to be adjusted with the new go-ipfs http-api
-    if (typeof cid === 'function') {
-      _callback = cid
-      cid = {}
+  return promisify((block, options, callback) => {
+    if (typeof options === 'function') {
+      callback = options
+      options = {}
     }
 
-    const callback = once(_callback)
```

Comment:
I can't remember but I wonder if this was a protection to go-ipfs sometimes sending multiple errors ?!

Reply:
Noted, I guess we'll find out soon ;)